src/NuGet.Clients/NuGet.SolutionRestoreManager.Interop/IVsSolutionRestoreService.cs

```diff
@@ -0,0 +1,29 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Runtime.InteropServices;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace NuGet.SolutionRestoreManager
+{
+    /// <summary>
+    /// Represents a package restore contract for integration with a project system.
+    /// </summary>
+    [ComImport]
+    [Guid("2b046428-ca39-40bb-8b4b-7dd1d96118cb")]
+    public interface IVsSolutionRestoreService
+    {
+        /// <summary>
+        /// This property will tell if last/current restore was a success or a failure
+        /// </summary>
+        Task<bool> CurrentRestoreOperation { get; }
```

Comment:
This task is a reflection of the current state of the current-restore-operation/ recently-completed-restore. The usage of this property will be to continue(to build solution or somthing) on completion of this task. Also, on completion, if the task returns false then it means the restore failed and the build task will be terminated.